### PR TITLE
fix: use def sites in `FindReferencesProvider`

### PR DESCRIPTION
**Pull request by nicdard**
Expand from https://github.com/flix/flix/issues/3337 and related PR.

@magnus-madsen just one doubt: Do I need to apply the fix also to Fields, Preds and Tags?

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/provider/FindReferencesProvider.scala [hunk 1]
```diff
@@ -75,26 +75,30 @@ object FindReferencesProvider {
   }
 
   private def findClassUses(sym: Symbol.ClassSym)(implicit index: Index, root: Root): JObject = {
+    val clazz = Location.from(sym.loc)
     val uses = index.usesOf(sym)
-    val locs = uses.toList.map(Location.from)
+    val locs = clazz :: uses.toList.map(Location.from)
     ("status" -> "success") ~ ("result" -> locs.map(_.toJSON))
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/FindReferencesProvider.scala:81]: Suggestion: Let us use the same two local variable names in all of these. Specifically i propose:

- "defSite"
- "useSites"

and then we always write `defSite :: useSites`.

Makes sense?

Comment (nicdard) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/FindReferencesProvider.scala:81]: sure totally :)

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/FindReferencesProvider.scala:81]: Very good. I think we should rename the functions like you did before too. E.g. `findClassUses` becomes `findClassReferences`. Agree?

Comment (nicdard) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/FindReferencesProvider.scala:81]: Oh yes you are completely true, I am sorry for that i completely forgot

Comment (nicdard) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/FindReferencesProvider.scala:81]: I updated as suggested, I have also fixed the references of tags :)

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**